Code/src/main/java/nl/utwente/viskell/ui/components/Block.java

```diff
@@ -100,17 +80,13 @@ public final CustomUIPane getPane() {
     /**
      * @return All InputAnchors of the block.
      */
-    public List<InputAnchor> getAllInputs() {
-        return ImmutableList.of();
-    }
+    public abstract List<InputAnchor> getAllInputs();
 
     /**
      * @return the optional output Anchor for this Block
      * TODO generalize to List<OutputAnchor> getOutputAnchors()
      */
-    public Optional<OutputAnchor> getOutputAnchor() {
-        return Optional.empty();
-    }
```

Comment:
I'd say these are reasonable defaults - I'm curious about your reasons for dropping them.